Changelog for pause & purge (#21134)
cloudflare/cloudflare-docs: src/content/changelog/queues/2025-03-25-pause-purge-queues.mdx

---

> title: Pause & Purge Queues

Reviewer: ```suggestion
title: New Pause & Purge APIs for Queues
```

> description: Manage downtime and invalid messages by pausing delivery, or deleting messages, from a queue.

Reviewer: Downtime is odd framing here - it sounds like a reliability issue?

```suggestion
description: Queues now supports the ability to pause delivery and/or delete messages from a Queue, allowing you to better manage queue backlogs.
```

products:
  - queues
date: 2025-03-27 12:00:00 UTC
---

> You can now pause message delivery from a [queue](/queues/), and purge all messages from a queue. These operations are useful to manage downtime, handle invalid messages, or run migrations.

Reviewer: ```suggestion
[Queues](/queues/) now supports the ability to pause message delivery and/or purge (delete) messages on a queue. These operations can be useful when:

* Your consumer has a bug or downtime and you want to stop messages from being processed temporarily
* If you've pushed invalid messages to a queue due to a code change during development, and need to clean up a queue without re-creating the queue
* Your queue has a backlog that is stale and you want to clean it up to allow new messages to be delivered
```

To pause a queue using [Wrangler](/workers/wrangler/), run the `pause-delivery` command. Paused queues continue to receive messages. And you can easily unpause a queue using the `resume-delivery` command.
```bash title="Pause and resume a queue"
$ wrangler queues pause-delivery my-queue
Pausing message delivery for queue my-queue.
Paused message delivery for queue my-queue.

$ wrangler queues resume-delivery my-queue
Resuming message delivery for queue my-queue.
Resumed message delivery for queue my-queue.
```

Purging a queue permanently deletes all messages in the queue. Unlike pausing, purging is an irreversible operation:
```bash title="Purge a queue"
$ wrangler queues purge my-queue
✔ This operation will permanently delete all the messages in queue my-queue. Type my-queue to proceed. … my-queue
Purged queue 'mar-21'
```

> You can also do these operations using our REST API, or the dashboard page for a queue.

Reviewer: Link to the REST API docs.

---
title: New Pause & Purge APIs for Queues
description: Queues now supports the ability to pause delivery and/or delete messages from a queue, allowing you to better manage queue backlogs.
products:
  - queues
date: 2025-03-27 12:00:00 UTC
---

[Queues](/queues/) now supports the ability to pause message delivery and/or purge (delete) messages on a queue. These operations can be useful when:

> * Your consumer has a bug or downtime, and you want to stop messages from being processed temporarily while you fix the bug

Reviewer 2: ```suggestion
* Your consumer has a bug or downtime, and you want to temporarily stop messages from being processed while you fix the bug
```

> * If you've pushed invalid messages to a queue due to a code change during development, and you want to clean up the backlog

Reviewer 2: ```suggestion
* You have pushed invalid messages to a queue due to a code change during development, and you want to clean up the backlog
```
Making it consistent with the first / last bullet point
